design/fips-compliance.md

```diff
@@ -0,0 +1,199 @@
+    which is one of the big players in the field of certifications, enterprise, 
+    and federal market.
+   - The pipeline should use and subscribe RHEL / UBI on-the-fly.
+   
+1. Use BoringSSL based crypto
+   Use [Google’s fork of Golang](https://github.com/golang/go/blob/dev.boringcrypto.go1.12/misc/boring/README.md) 
+   that uses dev.boringcrypto as its crypto engine. dev.boringcrypto wraps BoringSSL which is FIPS compliant.
+   
+Looking into the options above, it was clear that the best option is BoringSSL.
+Using cgo (option 1) was disqualified due to its complexity and consuming time
+ and effort as we will have to replace any usage of crypto.
+ 
+Using the RHEL go toolchain has its infrastructure complexity as we will need
+ to build a new pipeline for building the project on a subscribed RHEL machine.
+ 
+Luckily, Google maintains the BoringSSL which is very easy to use and gives
+ us FIPS compliancy with minimal work. All that needs to be done is:
+ - Replace the base image from `golang` to the corresponding `goboring/golang
+ `. For example, `golang:1.12` should turn into `goboring/golang:1.12.17b4`
+ - Enable cgo by setting `CGO_ENABLED=1` in the env
+ - Verify that the `authenticator` binary uses the BoringCrypto libraries by
+  running `go tool nm authenticator` and verifying that it has symbols named
+   *_Cfunc__goboringcrypto_*
+   
+More info on the required changes can be found [here](https://github.com/cyberark/conjur-authn-k8s-client/pull/97)
+
+### Non-FIPS compliant image
+
+We should decide if we still want to release a version of the `conjur-authn-k8s-client`
+that is not FIPS compliant (i.e that uses `golang` as the base image). Let's
+look into some pros and cons of both approaches.
+ 
+ | **Option**                             | **Implications**                                                                                                                                                                                                                                                                                                                                                        |
+ |----------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+ | Release only `goboring/golang` version | - Customers use the client and don't require FIPS compliancy, will now miss libraries that are available in `golang` but are not available in `goboring/golang`                                                                                                                                                                                                       |
```

Comment:
is there a significant amount of libraries not also available in `goboring/golang`?

Reply:
I am waiting for @izgeri on this? I didn't find them.